picard/config.py

```diff
@@ -70,43 +65,42 @@ def __getitem__(self, name):
         return self.value(name, opt, opt.default)
 
     def __setitem__(self, name, value):
-        key = self.__name + '/' + name
         self.lock_for_write()
         try:
-            self.__config[key] = value
-            self.__qt_config.setValue(key, value)
+            self.__qt_config.setValue(self.key(name), value)
         finally:
             self.unlock()
 
     def __contains__(self, name):
-        key = self.__name + '/' + name
-        return key in self.__config
+        return name in self._subkeys()
 
     def remove(self, name):
-        key = self.__name + '/' + name
         self.lock_for_write()
         try:
-            if key in self.__config:
-                self.__config.pop(key)
-                self.__qt_config.remove(key)
+            if name in self:
+                self.__qt_config.remove(self.key(name))
         finally:
             self.unlock()
 
-    def raw_value(self, name):
+    def raw_value(self, name, qtype=None):
         """Return an option value without any type conversion."""
-        value = self.__config[self.__name + '/' + name]
-        return value
+        key = self.key(name)
+        if qtype is not None:
+            return self.__qt_config.value(key, type=qtype)
+        else:
+            return self.__qt_config.value(key)
 
     def value(self, name, option_type, default=None):
         """Return an option value converted to the given Option type."""
-        key = self.__name + '/' + name
         self.lock_for_read()
         try:
-            if key in self.__config:
-                return option_type.convert(self.raw_value(name))
+            if name in self:
+                value = self.raw_value(name, qtype=option_type.qtype)
+                return option_type.convert(value)
             return default
-        except Exception:
-            log.error('Error reading option value', exc_info=True)
+        except Exception as why:
+            log.error('Cannot read %s value: %s', self.key(name), why)
+            callers(log.error, offset=2, length=2)
```

Comment:
Why `callers` instead of `exc_info`?

Reply:
Because `callers` lists callers... `exc_info` was too verbose, and wasn't really giving the useful info we need in this case.

Reply:
Ok, tested it, this somewhat makes sense. My main concern would be that we start cluttering the code with two calls, one for `log.error` and one for `callers`. At least when we start to use this elsewhere we should consider integrating this into the existing `picard.log` module somehow. For this PR I would be ok with merging this as is, though.

But can we at least log the exception itself with `%r`? Otherwise only the exception message is shown, but actually I think the type of exception is more interesting in many cases.

Also why limit the length to 2?

Reply:
I removed `callers` change from this PR, it can be done afterwards.